### PR TITLE
Bugfix/808 sync status

### LINKED ISSUE #808: new sync

### PR DESCRIPTION
**Pull request by matthewcarlreetz**
Closes #808

### CODE REVIEW

File: src/features/hotspots/details/HotspotDetails.tsx [hunk 1]
```diff
@@ -164,16 +166,10 @@ const HotspotDetails = ({
...
 
-    dispatch(
-      fetchSyncStatus({
-        address,
-        statusTime: hotspot?.status?.timestamp,
-        blockHeight,
-      }),
-    )
-  }, [address, blockHeight, dispatch, hotspot?.status?.timestamp])
+    dispatch(fetchSyncStatus({ hotspot, hotspotSyncBuffer }))
```
Comment (allenan) [inline src/features/hotspots/details/HotspotDetails.tsx:171]: feels a little odd for a view to know about sync buffer?

File: src/features/hotspots/details/HotspotStatusBanner.tsx [hunk 2]
```diff
@@ -5,40 +5,63 @@ import React, { useState, useEffect, memo, useMemo, useCallback } from 'react'
...
-  const hotspotSyncStatus = useMemo(() => syncStatuses[hotspot.address], [
-    hotspot.address,
-    syncStatuses,
-  ])
+
+  const syncStatusMessage = useMemo(() => {
+    if (!blockHeight || !syncStatus || hotspot.status?.online !== 'online')
+      return null
+
+    const timeAgo = hotspot.status?.timestamp
```
Comment (allenan) [inline src/features/hotspots/details/HotspotStatusBanner.tsx:45]: I'm seeing this logic repeat, can it be centralized somewhere? like a `<SyncStatusMessage />`?

File: src/utils/hotspotUtils.ts [hunk 3]
```diff
@@ -1,40 +1,36 @@
...
+  hotspotBlockHeight: number
+  blockHeight?: number
+  hotspotSyncBuffer?: number
+}): HotspotSyncStatus => {
+  if (!blockHeight || !hotspotSyncBuffer) {
+    return 'none'
+  }
 
   const syncedRatio = hotspotBlockHeight / blockHeight
   const percentSynced = round(syncedRatio * 100, 2)
```
Comment (allenan) [inline src/utils/hotspotUtils.ts:20]: I don't think the concept of percent synced really applies anymore. It's more like hotspot was synced/syncing/offline as of x time ago.

Comment (allenan) [reply src/utils/hotspotUtils.ts:20]: I guess maybe diagnostics could use the more traditional sync logic, but maybe that should be separate from the redux/gossip sync logic?

**Comment by allenan** [comment]
the logic throughout here feels pretty difficult to parse, guessing just how it's evolved. But it's pretty difficult to trace a line through what's happening and the function signature for `getSyncStatus` for example requires the caller to know a lot about the internal state of sync logic

File: src/features/hotspots/details/HotspotDetails.tsx (hunk 1 above)
Comment (matthewcarlreetz) [reply src/features/hotspots/details/HotspotDetails.tsx:171]: I agree. I guess I could pull it inside the slice from state. That's probably better.

**Comment by matthewcarlreetz** [comment]
> the logic throughout here feels pretty difficult to parse, guessing just how it's evolved. But it's pretty difficult to trace a line through what's happening and the function signature for `getSyncStatus` for example requires the caller to know a lot about the internal state of sync logic

Yeah, you're right. I'll try to refactor it for clarity.

**Review by allenan — APPROVED** [review]
nice refactor

**Pull request merged**